src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -367,6 +388,175 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 
 
 /*
+ * Fails if the expression contains STABLE functions which accept any paramaters derived
+ * from a Var. Assumes the expression contains no VOLATILE functions.
+ *
+ * Var's are allowed, but only if they are passed solely to IMMUTABLE functions
+ */
+static bool ContainsDisallowedFunctionCalls(Node *expression)
+{
+	bool unused = false;
+	return ContainsDisallowedFunctionCallsWalker(expression, &unused);
+}
+
+
+static bool ContainsDisallowedFunctionCallsWalker(Node *expression, bool *containsVar)
```

Comment:
Hm. This looks like it'll be painful to maintain across postgres versions.  Seems quite easy to miss new things to check when supporting a new postgres version.  Don't really have a better idea though. The only alternative I see is to only support a very restricted subset of expressions and automatically assume a disallowed function if an expression outside that set is encountered.